### ISSUE TITLE
Loading a revision loads revision class instead of revisable class

### ISSUE THREAD

**Issue by cblunt** [rich/acts_as_revisable#8]
If I have a revisable class `Person` and `PersonRevision`, any time I load a revised `Person`, the result is a `PersonRevision` class (instead of a `Person` class).

This is throwing my partial rendering and link generators. For example:

```
# app/models/person.rb
class Person
  acts_as_revisable
end

# app/models/person_revision.rb
class PersonRevision
  acts_as_revision
end

# app/controllers/people_controller.rb
@people = Person.all

# app/views/people/index.html.erb
<%= render @people -%>
```

For any revised person, the renderer will look for a partial `_person_revision.html.erb` instead of `_person.html.erb`.

Is this the expected behaviour; if so is there a automatic method to cast the revised `PersonRevision` to a `Person` class?

Thanks!
Chris

**Comment by cblunt** [comment]
Realised my mistake, I should explicitly set the link_to path in the partial, e.g.:
    <%= link_to "edit", person_path(person) -%>

However, the partial still loads the record as a PersonRevision class, which doesn't implement the instance methods of Person (e.g. `is_admin?`). I'm thinking of moving these methods into a module, and then including that module in `Person` and `PersonRevision`.

**Comment by cblunt** [comment]
I'm closing this issue as further looking into the code reveals the issue is caused by my revised model using the is_paranoid plugin alongside acts_as_revisable. I've not looked into  the source, but I'm guessing that the default_scopes are conflicting. I'll open a new issue once I've looked into it a little more.

**Comment by rich** [comment]
I'm not entirely sure there really is a problem here. Or at least the problem is one of documentation. 

What you're seeing is specifically how AAR is designed to work. You could handle this case by providing the partial Rails is looking for or, use the longer form of render like:

```
render :partial => 'people/person', :collection => @people
```

As for the is_paranoid issue, I'll look into that. One small note, it's not as fleshed out as the rest of the library but, there is some work at supporting is_paranoid type functionality.

```
acts_as_revisable :on_delete => :revise
```

That will flag and store the deleted record instead of deleting it. Then on your revision class there's simple a scope to access those records:

```
PersionRevision.deleted
```

**Comment by cblunt** [comment]
Thanks Rich. I looked a little more into the problem and confirmed that it was due to the limitation of ActiveRecord only allowing one `default_scope` to be declared.

Unlike `named_scope`, default scopes aren't merged, so is_paranoid was overwriting AAR's `default_scope` to only retrieve the current revision.

My `Person.find(:all)` was therefore finding every revision of every `Person`, so I had a mixture of `Person` and `PersonRevision` classes in the array.

I managed to solve the problem using a fork of `is_paranoid` http://github.com/grioja/is_paranoid which I found via this post http://joshuaclayton.github.com/code/default_scope/activerecord/is_paranoid/multiple-default-scopes.html
